core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -196,7 +198,37 @@ public RequestTemplate create(Object[] argv) {
           }
         }
       }
-      return resolve(argv, mutable, varBuilder);
+
+      RequestTemplate template = resolve(argv, mutable, varBuilder);
+
+      // add query map parameters after initial resolve so that they take
+      // precedence over any predefined values
+      return addQueryMapQueryParameters(argv, template);
+    }
+
+    private RequestTemplate addQueryMapQueryParameters(Object[] argv, RequestTemplate mutable) {
+    	for (int currQueryMapIndex : metadata.queryMapIndices()) {
+      	  Object value = argv[currQueryMapIndex];
+      	  if (value instanceof Map) {
```

Comment:
break the contract if the parameter is not a Map.. then you don't have to check it here